draft-annevk-johannhof-httpbis-cookies.md

```diff
@@ -1193,11 +1227,24 @@ boolean _httpOnlyAllowed_, boolean _allowNonHostOnlyCookieForPublicSuffix_, and
 
 1. Insert _cookie_ into the user agent's cookie store.
 
-1. Remove all expired cookies from the user agent's cookie store.
+1. Return _cookie_.
+
+
+### Garbage Collect Cookies
+
+To **Garbage Collect Cookies** given a host _host_:
 
-1. Run Remove Excess Cookies for Host given _cookie_'s host.
+1. Let _expiredCookies_ be a list containing all expired cookies in the user agent's cookie store.
 
-1. Run Remove Global Excess Cookies.
+1. For each _cookie_ in _expiredCookies_, remove _cookie_ from the user agent's cookie store.
+
+1. Let _excessHostCookies_ be the result of running Remove Excess Cookies for Host given _host_.
+
+1. Let _excessGlobalCookies_ be the result of running Remove Global Excess Cookies.
+
+1. Let _removedCookies_ be the concatenation of _expiredCookies_, _excessHostCookies_, and _excessGlobalCookies_.
```

Comment:
This is not an operation https://infra.spec.whatwg.org/#lists currently defines. Perhaps define _removedCookies_ upfront and use "extend"? We could also define list concatenation I suppose, might be useful elsewhere.